go-selinux/selinux_linux.go

```diff
@@ -1102,7 +1102,18 @@ func chcon(fpath string, label string, recurse bool) error {
 	}
 
 	if !recurse {
-		return setFileLabel(fpath, label)
+		err := lSetFileLabel(fpath, label)
+		if err == nil {
+			return nil
+		}
+		if errors.Is(err, os.ErrNotExist) {
+			return err
+		}
+		flabel, _ := lFileLabel(fpath)
+		if flabel == label {
+			return nil
+		}
+		return err
```

Comment:
Somewhat related to my earlier question; if reading back the label is faster than setting it, should we reverse this?

Currently, this;

- tries to set the label
- if that fails for _whatever reason_, other than "file doesn't exist", we try to read the existing label
- if that fails, we return the first error

If reading is indeed faster, we could do;


```suggestion
		if flabel, err := lFileLabel(fpath); err == nil && flabel == label {
			// already has the right label
			return nil
		}

		return lSetFileLabel(fpath, label)
```